### ISSUE TITLE
Duplicate wp.media popup dialog markup when add/edit shortcode.

### ISSUE THREAD

**Issue by rahulsprajapati** [wp-shortcake/shortcake#679]
When we open wp.media model on `.shortcake-add-post-element` button click event or edit the shortcode event we are creating new media model every time. We should either use the existing model if its exist or we should remove existing model. 
Because currently it is adding duplicate markup, like we have one element with id '**inner_content**' for shortcode content which we can't access by id if we have more than one element with same id. 

Here for Edit Shortcode : https://github.com/wp-shortcake/shortcake/blob/master/js/src/utils/shortcode-view-constructor.js#L169-L175
This should be 
```
if ( wp.media.frames.wp_media_frame ) {
	wp.media.frames.wp_media_frame.remove();
}

if ( wp.media.frames.wp_media_new_frame ) {
	wp.media.frames.wp_media_new_frame.remove();
}

var wp_media_frame = wp.media.frames.wp_media_frame = wp.media({
	frame : "post",
	state : 'shortcode-ui',
	currentShortcode : currentShortcode,
});

wp_media_frame.open();
```
OR we can use existing one 
```
if ( wp.media.frames.wp_media_new_frame ) {
	wp.media.frames.wp_media_new_frame.remove();
	wp.media.frames.wp_media_frame = '';
}

if ( wp.media.frames.wp_media_frame ) {
	wp_media_frame = wp.media.frames.wp_media_frame;
} else {
	wp_media_frame = wp.media.frames.wp_media_frame = wp.media({
		frame : "post",
		state : 'shortcode-ui',
		currentShortcode : currentShortcode,
	});
}
wp_media_frame.open();
```
And for `Add Post Element` https://github.com/wp-shortcake/shortcake/blob/master/js/src/shortcode-ui.js#L42-L43
I think we are trying to remove editor by using `wp.media.editor.remove( editor );` but it will not work as the parameter `editor` is target content editor id.
I think this should be :
```
if ( wp.media.frames.wp_media_frame ) {
	wp.media.frames.wp_media_frame.remove();
}
if ( wp.media.frames.wp_media_new_frame ) {
	wp.media.frames.wp_media_new_frame.remove();
}
var wp_media_new_frame = wp.media.frames.wp_media_new_frame = wp.media(options);
wp_media_new_frame.open();
```
OR we can use existing one
```
if ( wp.media.frames.wp_media_frame ) {
	wp.media.frames.wp_media_frame.remove();
	wp.media.frames.wp_media_frame = '';
}
var wp_media_new_frame;
if ( wp.media.frames.wp_media_new_frame ) {
	wp_media_new_frame = wp.media.frames.wp_media_new_frame;
} else {
	wp_media_new_frame = wp.media.frames.wp_media_new_frame = wp.media(options);
}
wp_media_new_frame.open();
```

**Comment by rahulsprajapati** [comment]
@goldenapples @mattheu 
Please give your thought about this. If you feel this is right way I can raise the PR for this.

**Comment by goldenapples** [comment]
I think would be a very good fix to get in here. 

I think I prefer the second option you're proposing in each case - if we create a new media frame rather than reusing the existing one, there's still a few nodes of extra markup that don't get cleaned up on the page. It looks like the markup gets removed from the modal, which solves the main problem you were addressing, but the `.media-modal` div is still left hanging around, as this screenshot shows:

<img width="822" alt="screen shot 2016-12-27 at 10 23 19 am" src="https://cloud.githubusercontent.com/assets/665992/21505711/ae9869f4-cc1e-11e6-8449-669483b24816.png">

So unless there are other considerations, I'd like to use that option.

One thing that looks possibly unnecessary about the code you posted: is it necessary to set `wp.media.frames.wp_media_frame` to an empty string? That seems to cause the same duplicate markup issue by causing a new media frame model to be initialized on each interaction. I could be missing something, but your code worked fine for me without that line.

I'd definitely welcome a PR for this. I believe issues like this have been brought up as bugs here before, but I've never tracked down the source of the bugs, and I very much appreciate you doing so!

**Comment by rahulsprajapati** [comment]
@goldenapples 
yes, we have to clean the variable `wp.media.frames.wp_media_frame`  when we are removing the model. Because we are checking this var to find the existing model. 
```
if ( wp.media.frames.wp_media_frame ) {
	wp_media_frame = wp.media.frames.wp_media_frame;
} else {
	wp_media_frame = wp.media.frames.wp_media_frame = wp.media({
		frame : "post",
		state : 'shortcode-ui',
		currentShortcode : currentShortcode,
	});
}
```
If we don't clean this var than it may open blank media model. So we should empty the var when we are removing the model.

**Comment by goldenapples** [comment]
Closed via #689.
